src/index/mod.rs

```diff
@@ -70,6 +71,7 @@ pub struct Index {
     delete: Arc<Delete>,
     protect: Mutex<IndexProtect>,
     view: ArcSwap<IndexView>,
+    optimize_unparker: OnceLock<Unparker>,
```

Comment:
You need to read the document and examples of parker. What you implemented before is just a parker.

Knowing if indexing is building is still needed.

Reply:
done in new commit

Reply:
Why OnceLock?

Why unpark is called by optimizing thread?

Reply:
> Why OnceLock?

The optimizing thread will be created after index is created. So we need `OnceLock` to lazy set the unparker.

> Why unpark is called by optimizing thread?

Unpark will be called by both refresh and optimizing thread. Unpark in optimizing thread is used to clean the exist token, otherwise the park operation will be unparked by refresh function called before starting optimizing.

Reply:
> Why unpark can be used to clean the existing token?

Because `unpark` won't remember previous token. No mater what previous state is, after doing `u.unpark(); p.park();`, the token will be cleared.

Reply:
Why clear the token?

Reply:
```
refresh
start optimizing
done -> park
```

If events happen with this order, the park function will return immediately and it will only be parked in next loop.

Reply:
enter `optimizing_indexing` function

Reply:
Is not the token consumed by line 436?

Reply:
If refresh function is called again during optimizing, it won't be parked.

Reply:
I see. I do not think it is a problem. Compared to time-consuming building index, checking if an index is needed is not so important.

Reply:
Removed in new commit.